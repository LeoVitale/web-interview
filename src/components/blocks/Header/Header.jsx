import React from 'react';
import MenuItem from 'components/blocks/MenuItem';
import Avatar from 'components/blocks/Avatar';
import styles from  './Header.module.scss';

const Header = ({ logo }) => {
  const { appHeader, appLogo, menu } = styles;
  return (
    <header className={appHeader}>
      <img src={logo} className={appLogo} alt="logo" />
      <nav className={menu}>
        <MenuItem label="Book" to="/new-appointments" />
        <MenuItem label="Appointments" to="/" />
        <MenuItem label="Family Members" to="/family-member" />
        <Avatar />
      </nav>
    </header>
  )
}

export default Header;
