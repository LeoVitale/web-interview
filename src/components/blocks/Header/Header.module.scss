@import 'styles/_constants.scss';

.appHeader {
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: $gutter-size--large;

    .appLogo {
      max-height: 33px;
    }
    border-bottom: $border-standard;

    .menu {
      display: flex;
      align-items: center;
    }
  }
