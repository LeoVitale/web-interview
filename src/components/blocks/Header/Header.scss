@import '../../../constants';

.app-header {
    padding: $gutter-size--large;

    .app-logo {
      max-height: 33px;
    }
    border-bottom: $border-standard;
  }
