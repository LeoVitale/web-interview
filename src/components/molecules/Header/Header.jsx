import React from 'react'
import PropTypes from 'prop-types'
import MenuItem from 'components/atoms/MenuItem'
import Avatar from 'components/atoms/Avatar'
import styles from './Header.module.scss'

const Header = ({ logo }) => {
  const { appHeader, appLogo, menu } = styles
  return (
    <header className={appHeader}>
      <img src={logo} className={appLogo} alt="logo" />
      <nav className={menu}>
        <MenuItem label="Book" to="/new-appointments" />
        <MenuItem label="Appointments" to="/" />
        <MenuItem label="Family Members" to="/family-members" />
        <Avatar size="small"/>
      </nav>
    </header>
  )
}

Header.propTypes = {
  logo: PropTypes.string.isRequired,
}

export default Header
