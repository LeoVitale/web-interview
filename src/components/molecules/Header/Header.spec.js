import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import { mount } from 'enzyme'
import Header from './Header.jsx'

describe('<Header />', () => {
  const header = mount(
    <Router>
      <Header />
    </Router>
  )

  it('should render Header as expected', () => {
    expect(header).toMatchSnapshot()
  })
})
