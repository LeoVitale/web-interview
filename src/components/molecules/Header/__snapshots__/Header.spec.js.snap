// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render Header as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Header>
      <header
        className="appHeader"
      >
        <img
          alt="logo"
          className="appLogo"
        />
        <nav
          className="menu"
        >
          <MenuItem
            className=""
            icon="book"
            label="Book"
            to="/new-appointments"
          >
            <Link
              className="menuItem "
              replace={false}
              to="/new-appointments"
            >
              <a
                className="menuItem "
                href="/new-appointments"
                onClick={[Function]}
              >
                <Icon
                  className="iconItem"
                  type="book"
                >
                  <svg
                    className="svgIcon iconItem"
                    fill="#400099"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <title />
                    <path
                      d="M19.504 17.997h1.75a.75.75 0 0 1 0 1.5h-1.75v1.75a.75.75 0 1 1-1.5 0v-1.75h-1.75a.75.75 0 1 1 0-1.5h1.75v-1.75a.75.75 0 0 1 1.5 0v1.75zM3.527 8.25h17.05V5a.25.25 0 0 0-.25-.25h-3.32v1.092a.75.75 0 1 1-1.5 0V4.75H8.441v1.038a.75.75 0 0 1-1.5 0V4.75H3.777a.25.25 0 0 0-.25.25v3.25zm17.05 1.5H3.527V19c0 .138.112.25.25.25h9.526a.75.75 0 1 1 0 1.5H3.777A1.75 1.75 0 0 1 2.027 19V5c0-.966.784-1.75 1.75-1.75h3.164v-.449a.75.75 0 1 1 1.5 0v.449h7.066v-.449a.75.75 0 1 1 1.5 0v.449h3.32c.966 0 1.75.784 1.75 1.75v9.601a.75.75 0 1 1-1.5 0V9.75z"
                    />
                  </svg>
                </Icon>
                <span>
                  Book
                </span>
              </a>
            </Link>
          </MenuItem>
          <MenuItem
            className=""
            icon="appointments"
            label="Appointments"
            to="/appointments"
          >
            <Link
              className="menuItem "
              replace={false}
              to="/appointments"
            >
              <a
                className="menuItem "
                href="/appointments"
                onClick={[Function]}
              >
                <Icon
                  className="iconItem"
                  type="appointments"
                >
                  <svg
                    className="svgIcon iconItem"
                    fill="#400099"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <title />
                    <path
                      d="M3.777 4.75a.25.25 0 0 0-.25.25v14c0 .138.112.25.25.25h16.55a.25.25 0 0 0 .25-.25V5a.25.25 0 0 0-.25-.25H3.777zm0-1.5h16.55c.966 0 1.75.784 1.75 1.75v14a1.75 1.75 0 0 1-1.75 1.75H3.777A1.75 1.75 0 0 1 2.027 19V5c0-.966.784-1.75 1.75-1.75zm3.164-.449a.75.75 0 1 1 1.5 0v2.987a.75.75 0 0 1-1.5 0V2.8zm8.566 0a.75.75 0 1 1 1.5 0v3.04a.75.75 0 1 1-1.5 0v-3.04zM2.75 9.75v-1.5h18.5v1.5H2.75zm12.5 4v2.5h2.5v-2.5h-2.5zm-.75-1.5h4a.75.75 0 0 1 .75.75v4a.75.75 0 0 1-.75.75h-4a.75.75 0 0 1-.75-.75v-4a.75.75 0 0 1 .75-.75z"
                    />
                  </svg>
                </Icon>
                <span>
                  Appointments
                </span>
              </a>
            </Link>
          </MenuItem>
          <MenuItem
            className=""
            icon="family"
            label="Family Members"
            to="/family-members"
          >
            <Link
              className="menuItem "
              replace={false}
              to="/family-members"
            >
              <a
                className="menuItem "
                href="/family-members"
                onClick={[Function]}
              >
                <Icon
                  className="iconItem"
                  type="family"
                >
                  <svg
                    className="svgIcon iconItem"
                    fill="#400099"
                    height="24"
                    role="img"
                    viewBox="0 0 24 24"
                    width="24"
                  >
                    <title />
                    <path
                      d="M17 12.75a2.75 2.75 0 1 1 0-5.5 2.75 2.75 0 0 1 0 5.5zm0-1.5a1.25 1.25 0 1 0 0-2.5 1.25 1.25 0 0 0 0 2.5zm1.1 3.898a.75.75 0 0 1 .8-1.268c1.307.825 2.868 3.612 3.105 5.465a.75.75 0 0 1-.743.845h-2.334a.75.75 0 1 1 0-1.5h1.39c-.448-1.385-1.48-3.075-2.219-3.542zm-3.007 3.547c-.55-2.889-2.378-4.26-5.665-4.26s-5.162 1.374-5.793 4.26h11.458zm-13.09.655c.535-4.218 3.084-6.416 7.425-6.416 4.348 0 6.85 2.208 7.276 6.436a.75.75 0 0 1-.747.825H2.747a.75.75 0 0 1-.745-.845zm7.409-7.284c-2.227 0-4.044-1.743-4.044-3.908 0-2.166 1.817-3.908 4.044-3.908 2.226 0 4.044 1.742 4.044 3.908 0 2.165-1.818 3.908-4.044 3.908zm0-1.5c1.412 0 2.544-1.086 2.544-2.408 0-1.323-1.132-2.408-2.544-2.408C8 5.75 6.868 6.835 6.868 8.158c0 1.322 1.132 2.408 2.544 2.408z"
                    />
                  </svg>
                </Icon>
                <span>
                  Family Members
                </span>
              </a>
            </Link>
          </MenuItem>
        </nav>
      </header>
    </Header>
  </Router>
</BrowserRouter>
`;
